parsec/scheduling.c

```diff
@@ -796,3 +897,53 @@ int parsec_context_wait( parsec_context_t* context )
     context->flags ^= (PARSEC_CONTEXT_FLAG_COMM_ACTIVE | PARSEC_CONTEXT_FLAG_CONTEXT_ACTIVE);
     return (ret >= 0)? PARSEC_SUCCESS: ret;
 }
+
+int parsec_taskpool_wait( parsec_taskpool_t* tp )
+{
+    int ret = 0;
+    parsec_context_t *context = tp->context;
+
+    if( NULL == context ) {
+        parsec_warning("taskpool is not registered with any context in parsec_taskpool_wait\n");
+        return -1;
+    }
+
+    if( !(PARSEC_CONTEXT_FLAG_CONTEXT_ACTIVE & context->flags) ) {
+        parsec_warning("taskpool is registered to non-started context in parsec_taskpool_wait\n");
+        return -1;
+    }
+
+    if( __parsec_context_cas_or_flag(context,
+                                     PARSEC_CONTEXT_FLAG_COMM_ACTIVE) ) {
+        (void)parsec_remote_dep_on(context);
+    }
+
+    ret = __parsec_taskpool_wait( tp, context->virtual_processes[0]->execution_streams[0] );
+
+    return ret;
+}
+
+int parsec_taskpool_test( parsec_taskpool_t* tp )
+{
+    int ret = 0;
+    parsec_context_t *context = tp->context;
+
+    if( NULL == context ) {
+        parsec_warning("taskpool is not registered with any context in parsec_taskpool_wait\n");
+        return -1;
+    }
+
+    if( !(PARSEC_CONTEXT_FLAG_CONTEXT_ACTIVE & context->flags) ) {
+        parsec_warning("taskpool is registered to non-started context in parsec_taskpool_wait\n");
+        return -1;
+    }
+
+    if( __parsec_context_cas_or_flag(context,
+                                     PARSEC_CONTEXT_FLAG_COMM_ACTIVE) ) {
+        (void)parsec_remote_dep_on(context);
+    }
+
+    ret = __parsec_taskpool_test( tp, context->virtual_processes[0]->execution_streams[0] );
```

Comment:
This is the only call site, and the `es` argument is set to a "constant" that can be deduced from `tp->context`; is the second argument to `__parsec_taskpool_test` needed?

Reply:
Probably it's a place where we should use `parsec_my_execution_stream()`

Reply:
done.